### PR TITLE
feat(caldav): create subscriptions via occ

### PR DESCRIPTION
**Pull request by miaulalala**
## Summary

Allow adding subscriptions via OCC

## TODO

- [ ] Docs

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — CHANGES_REQUESTED** [review]
Inject the CalDavBackend to get rid of all the service locators

File: apps/dav/lib/Command/CreateSubscription.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class CreateSubscription extends Command {
+	public function __construct(
+		protected IUserManager $userManager,
+		private CalDavBackend $caldav,
+		protected IDBConnection $dbConnection,
```
Comment (ChristophWurst) [inline apps/dav/lib/Command/CreateSubscription.php:24]: Appears to be unused

**Review by ChristophWurst — APPROVED** [review]
Looks good

**Review by Altahrim — APPROVED** [review]

**Review by tcitworld — COMMENTED** [review]
Could be nice to execute subscription sync to test is the URL is valid?

File: apps/dav/lib/Command/CreateSubscription.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$user = $input->getArgument('user');
+		if (!$this->userManager->userExists($user)) {
+			$output->writeln("<error>User <$user> in unknown.</error>");
+			return self::FAILURE;
+		}
+
+		$name = $input->getArgument('name');
+		$url = $input->getArgument('url');
+		$color = $input->getArgument('color') ?? '#0082c9';
```
[the remark this answers is not in the document]
Comment (miaulalala) [reply apps/dav/lib/Command/CreateSubscription.php:54]: yeah I didn't really know what else to use here - we could move it to a config, we could use a random value... IDK if subscriptions have a default value on creation?

[the remark this answers is not in the document]
Comment (miaulalala) [reply apps/dav/lib/Command/CreateSubscription.php]: Using the theming defaults now

**Comment by miaulalala** [comment]
> Could be nice to execute subscription sync to test is the URL is valid?

I did have that initially, but since it's quite a lot of code duplication I desisted from doing that as it's not really neccessary. The RefreshWebCal Job will log if the url isn't valid.

**Pull request merged**